shotgun_api3/lib/mockgun/mockgun.py

```diff
@@ -300,23 +319,20 @@ def find(self, entity_type, filters, fields=None, order=None, filter_operator=No
                 else:
                     raise ValueError("Unknown ordering direction")
 
-                results = sorted(results, key=lambda k: k[order_field], reverse=desc_order)
+                val = sorted(val, key=lambda k: k[order_field], reverse=desc_order)
 
-        if fields is None:
-            fields = set(["type", "id"])
-        else:
-            fields = set(fields) | set(["type", "id"])
-
-        # get the values requested
-        val = [dict((field, self._get_field_from_row(entity_type, row, field)) for field in fields) for row in results]
+        # Remove any fields that was not explicitely requested.
+        fields_to_remove = all_fields - requested_fields
+        for v in val:
+            for field in fields_to_remove:
+                v.pop(field)
```

Comment:
Ah, never noticed that `dict` had a `pop` method just like `list`. Looks nicer than using `del v[field]`. You get the result back as well.

Nice!